src/nvim/lua/vim.lua

```diff
@@ -195,6 +187,13 @@ deepcopy = function(orig)
   return deepcopy_funcs[type(orig)](orig)
 end
 
+local function __index(table, key)
+  if key == "inspect" then
+    table.inspect = require("vim.inspect")
+    return table.inspect
```

Comment:
does `require()` do its own caching? If not, shouldn't we avoid it after `table.inspect` is non-nil?

Reply:
@justinmk the line above does the caching.

Reply:
Thanks. For [reference](https://stackoverflow.com/a/18026617/):

> `__index` is only accessed when the key being read from the main table does not already exist in the main table

When the key being read from the main table does not already exist in the main table. 

Also [this](https://stackoverflow.com/a/34235619/) explains `require()`:

> `require` caches loaded modules. Calling it again with the same module name will simply return the module that was previously loaded

Reply:
section 2.8 in https://www.lua.org/manual/5.1/manual.html contains functionalistic descriptions for how the metatable mechanism works.